### PR TITLE
Wires the PIP alarm to the alarm banner

### PR DESCRIPTION
**Pull request by jkff**
This is based on @paulovap 's https://github.com/RespiraWorks/VentilatorSoftware/pull/523 - I took that PR, rebased it on master, and tried to wire things together so that an alarm banner would appear when I change the commanded PIP to be low enough.

This now works as expected:
- you can tweak commanded PIP to activate or inactivate the alarm in fake mode
- when it activates, the banner appears
- when you press pause, the banner disappears, and reappears if the condition is still active 2 minutes later
- if condition goes down during silencing, that clears the silencing so if condition reappears then it's a new banner

Next step for the alarm subsystem are:
- wire things up with the "silenced alarms" widget
- add alarm settings GUI
- add more alarms and test them with real hardware

### CODE REVIEW

File: gui/app/main.qml [hunk 1]
```diff
@@ -69,6 +70,17 @@ ApplicationWindow {
...
         }
+
+        AlarmNotificationBanner {
+            id: alarmNotificationBanner
+            Component.onCompleted: console.log(GuiStateContainer.alarmManager.highestPriorityAlarm.isValid)
+            enabled: GuiStateContainer.alarmManager.highestPriorityAlarm.isAudioActive
+            priority: GuiStateContainer.alarmManager.highestPriorityAlarm.priority
+            title: GuiStateContainer.alarmManager.highestPriorityAlarm.text
+            alarmsCount: GuiStateContainer.alarmManager.numActiveAlarms
+            / onPauseAlarmClicked: GuiStateContainer.alarmManager.AcknowledgeHighestPriorityAlarm
```
[the remark this answers is not in the document]
Comment (paulovap) [reply gui/app/main.qml:81]: Also `AcknowledgeHighestPriorityAlarm` must start with lowercase for qml to understand (that is why I prefer Qt style over google style for UI)

**Comment by jkff** [comment]
Thanks Paulo! I've updated the PR using your changes + some more stuff, and it now works as expected.

File: gui/app/main.qml [hunk 2]
```diff
@@ -59,7 +60,7 @@ ApplicationWindow {
             }
 
             onMenuClicked: console.log("Menu Click")
-            onAlarmSettingsClicked: console.log("Alarm Settings Clicked")
+            onAlarmSettingsClicked: alarmNotificationBanner.enabled = true
```
Comment (paulovap) [inline gui/app/main.qml:63]: This was just for debug purpose. Are you planning to merge with this?

File: gui/app/Style.qml [hunk 3]
```diff
@@ -47,22 +47,52 @@ QtObject {
...
+              } else {
+                return "transparent";
+              }
             }
 
             property var timeSeries: QtObject {
               property var lineByPriority: function(p) {
                 var c = Style.theme.color
-                return (p == AlarmPriority.HIGH) ? c.alarmHighBright : c.graphLine;
+                if (p == AlarmPriority.HIGH) {
```
[the remark this answers is not in the document]
Comment (jkff) [reply gui/app/Style.qml:75]: Fixed.

File: gui/app/main.qml (hunk 2 above)
Comment (jkff) [reply gui/app/main.qml]: Fixed

**Pull request merged**